workflows/Snakefile

```diff
@@ -13,33 +17,33 @@ params_md["fnr"] = 0.097
 params_md["na_rate"] = 0.014
 params_md["observe_homozygous"] = ["False", "True"]
 
+# MCMC parameters of mark00
+params_mcmc = dict()
+params_mcmc["seed"] = 42
+params_mcmc["n_trees"] = 1
+params_mcmc["n_cells"] = 2000
+params_mcmc["n_mutations"] = 50
+params_mcmc["strategy"] = """UNIFORM_INCLUDE_ROOT"""
+params_mcmc["fpr"] = 1.24*10**(-6)
+params_mcmc["fnr"] = 0.097
+params_mcmc["na_rate"] = 0.014
+params_mcmc["observe_homozygous"] = "False"
+
 
 rule all:
     """Run all."""
     input:
-        mock_data = expand('../data/mock/seed_{seed}_n_trees_{n_trees}_n_cells_{n_cells}_n_mutations_{n_mutations}_fpr_{fpr}_fnr_{fnr}_na_rate_{na_rate}_observe_homozygous_{observe_homozygous}_strategy_{strategy}_tree_{tree_no}.json',
-                            seed=params_md["seed"],
-                            n_trees=params_md["n_tree"],
-                            n_cells=params_md["n_cells"],
-                            n_mutations=params_md["n_mutations"],
-                            strategy=params_md["strategy"],
-                            fpr=params_md["fpr"],
-                            fnr=params_md["fnr"],
-                            na_rate=params_md["na_rate"],
-                            observe_homozygous=params_md["observe_homozygous"],
-                            tree_no= list(range(1, params_md["n_tree"]+1))
-                        ),
-        mcmc_log_mark0ß = '../data/mcmc/mark00/mcmc_run.log',
-        mcmc_samples_mark00 = '../data/mcmc/mark00/samples.json'
-
-
+        mcmc_samples = "Test.json"
 
 rule make_random_or_deep_tree:
     """Make a tree (TreeNode) of random/deep topology and save it as JSON."""
     input:
         script="../scripts/make_tree.py"
+
+    wildcard_constraints:
```

Comment:
I thought that `expand` for the experimental design would actually be able to understand constraints, but I may be wrong

Reply:
Oh, my idea is to use expand later for my experimental specification. All these rules you see here provide snakemake the most general way to produce all files.